hydrachain/tests/test_working_app.py

```diff
@@ -0,0 +1,165 @@
+        while True:
+            self.log('wait_for_new_block')
+            block_hashes = self.client.call('eth_getFilterChanges', self.new_block_filter_id)
+            if block_hashes:
+                return block_hashes[0]
+            if timeout and time.time() - start_ts > timeout:
+                return None
+            time.sleep(0.5)
+
+    def connect_client(self):
+        while True:
+            try:
+                self.client = JSONRPCClient()
+                self.client.call('web3_clientVersion')
+                break
+            except ConnectionError, e:
+                time.sleep(0.5)
+
+    def run(self):
+        # Stdin is grabbed by CLIRunner so logs are stored internally
+        self.test_output = []
+        self.test_successful = False
+        self.log('test started')
+
+        try:
+            self.connect_client()
+            self.log('client connected')
+
+            # Set up filter to get notified when a new block arrives
+            self.new_block_filter_id = self.client.call('eth_newBlockFilter')
+            self.log('eth_newBlockFilter OK')
+
+            # Read initial blocks created by HydraChain on startup
+            while self.wait_for_new_block(timeout=3):
+                pass
+
+            # Create a contract
+            params = {'from': self.client.coinbase.encode('hex'),
+                      'to': '',
+                      'data': contract_code,
+                      'gasPrice': '0x{}'.format(self.gasprice)}
+            self.client.call('eth_sendTransaction', params)
+            self.log('eth_sendTransaction OK')
+
+            # Wait for new block
+            recent_block_hash = self.wait_for_new_block()
+            self.log('recent_block_hash {}'.format(recent_block_hash))
+
+            recent_block = self.client.call('eth_getBlockByHash', recent_block_hash, True)
+            self.log('eth_getBlockByHash OK {}'.format(recent_block))
+
+            assert recent_block['transactions'], 'no transactions in block'
+            tx = recent_block['transactions'][0]
+            assert tx['to'] == '0x'
+            assert tx['gasPrice'] == params['gasPrice']
+            assert len(tx['input']) > len('0x')
+            assert tx['input'].startswith('0x')
+
+            # Get transaction receipt to have the address of contract
+            receipt = self.client.call('eth_getTransactionReceipt', tx['hash'])
+            self.log('eth_getTransactionReceipt OK {}'.format(receipt))
+
+            assert receipt['transactionHash'] == tx['hash']
+            assert receipt['blockHash'] == tx['blockHash']
+            assert receipt['blockHash'] == recent_block['hash']
+
+            # Get contract address from receipt
+            contract_address = receipt['contractAddress']
+            code = self.client.call('eth_getCode', contract_address)
+            self.log('eth_getCode OK {}'.format(code))
+
+            assert code.startswith('0x')
+            assert len(code) > len('0x')
```

Comment:
Couldn't we tighten this assertion a little bit? `code` could be almost anything... ;)